plugins/utils/auto-session.nix

```diff
@@ -0,0 +1,187 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.auto-session;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.auto-session =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "auto-session";
+
+      package = helpers.mkPackageOption "auto-session" pkgs.vimPlugins.auto-session;
+
+      logLevel =
+        helpers.defaultNullOpts.mkEnum
+        ["debug" "info" "warn" "error"]
+        "error"
+        "Sets the log level of the plugin.";
+
+      autoSession = {
+        enable = helpers.defaultNullOpts.mkBool true ''
```

Comment:
Default is nil. Do we leave `enable` when the upstream option is `enabled` ?

Reply:
Default is actually `true`.
Renamed to `enabled`.